plugins/state_monitor.py

```diff
@@ -0,0 +1,238 @@
+"""
+Plugin to monitor state variables from the charge controller.
+"""
+
+# Standard library imports
```

Comment:
```suggestion
# Standard library imports
import abc  # For the MessageSender ABC
import logging  # For getting the logger from SlackSender
```

Add `abc` import needed below (if you accept my suggestion on that) and `logging`